packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1100,17 +1123,15 @@ export class GarbageCollector implements IGarbageCollector {
         }
 
         const gcSnapshotTree = snapshot.trees[gcTreeKey];
-        if (gcSnapshotTree !== undefined && this.trackGCState) {
+        if (gcSnapshotTree !== undefined) {
             const latestGCData = await getGCDataFromSnapshot(
                 gcSnapshotTree,
                 readAndParseBlob,
             );
-            this.latestSummaryData = {
-                serializedGCState: JSON.stringify(generateSortedGCState(latestGCData.gcState)),
-                serializedTombstones: JSON.stringify(latestGCData.tombstones),
-            };
+            this.updateStateFromSnapshotData(latestGCData);
         } else {
             this.latestSummaryData = undefined;
+            this.gcDataFromLastRun = undefined;
```

Comment:
Makes me wonder if we are refreshing from the latest summary, and `gcDataFromLastRun` is undefined, should we also clear the `unreferencedNodesState` as well.

Reply:
You mean if the gcSnapshotTree is not present. Good point, we should clean everything up.

Reply:
Updated the code to reset all states when gcSnapshotTree is not present.